Source/WebCore/svg/SVGElement.cpp

```diff
@@ -197,6 +197,18 @@ SVGElementRareData& SVGElement::ensureSVGRareData()
     return *m_svgRareData;
 }
 
+bool SVGElement::isInnerSVGSVGElement() const
+{
+    if (!is<SVGSVGElement>(*this))
+        return false;
+
+    // Element may not be in the document, pretend we're outermost for viewport(), getCTM(), etc.
+    if (!parentNode())
+        return false;
+
+    return dynamicDowncast<SVGElement>(parentNode());
```

Comment:
Is this as efficient as is<SVGElement>(svgElement.parentNode()) ?

Reply:
Better than is + downcast.

Reply:
Wrong. There is no need for is + downcast here, we only need is, hence is is prefered.d

Reply:
Fixed.